node/full_node_test.go

```diff
@@ -54,16 +52,14 @@ func TestMempoolDirectly(t *testing.T) {
 	assert := assert.New(t)
 	require := require.New(t)
 
-	app := &mocks.Application{}
-	app.On(InitChain, mock.Anything).Return(abci.ResponseInitChain{})
-	app.On(CheckTx, mock.Anything).Return(abci.ResponseCheckTx{})
-	key, _, _ := crypto.GenerateEd25519Key(rand.Reader)
-	signingKey, _, _ := crypto.GenerateEd25519Key(rand.Reader)
-	anotherKey, _, _ := crypto.GenerateEd25519Key(rand.Reader)
+	app := setupMockApplication()
+	key, signingKey := generateKeys()
+	anotherKey := generateSingleKey()
 
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
-	node, err := newFullNode(ctx, config.NodeConfig{DALayer: "mock"}, key, signingKey, proxy.NewLocalClientCreator(app), &types.GenesisDoc{ChainID: "test"}, log.TestingLogger())
+
+	node, err := setupNode(ctx, key, signingKey, app)
 	require.NoError(err)
 	require.NotNil(node)
```

Comment:
I would pass require into the helper and validate this in there so that this check is inherit and isn't duplicated in every test.

Reply:
I can't comment on the lines specifically, but you could also include the `node.Start()` in the helper function as well. 

If there are tests that need access to the node prior to starting then you could have 2 helpers

```go
func setupNode() { }

func setupAndStartNode() {
    node, err := setupNode()
    node.Start()
}
```